src/tests/ftest/util/dmg_utils.py

```diff
@@ -384,33 +391,33 @@ def pool_create(self, scm_size, uid=None, gid=None, nvme_size=None,
         }
         if target_list is not None:
             kwargs["ranks"] = ",".join([str(target) for target in target_list])
-        self._get_result(("pool", "create"), **kwargs)
+        self._get_json_result(("pool", "create"), **kwargs)
 
         # Extract the new pool UUID and SVC list from the command output
         data = {}
-        if self.json.value:
-            # Sample json output.
-            # "response": {
-            #     "UUID": "ebac9285-61ec-4d2e-aa2d-4d0f7dd6b7d6",
-            #     "Svcreps": [
-            #     0
-            #     ]
-            # },
-            # "error": null,
-            # "status": 0
-            output = json.loads(self.result.stdout)
-            data["uuid"] = output["response"]["UUID"]
-            data["svc"] = ",".join(
-                [str(svc) for svc in output["response"]["Svcreps"]])
-
-        else:
-            match = re.findall(
-                r"UUID:\s+([A-Za-z0-9-]+),\s+"
-                r"Service replicas:\s+([0-9]+(,[0-9]+)*)",
-                self.result.stdout)
-            if match:
-                data["uuid"] = match[0][0]
-                data["svc"] = match[0][1]
+        # Sample json output.
+        # "response": {
+        #   "uuid": "ebac9285-61ec-4d2e-aa2d-4d0f7dd6b7d6",
+        #   "svc_reps": [
+        #     0
+        #   ],
+        #   "tgt_ranks": [
+        #     0,
+        #     1
+        #   ],
+        #   "scm_bytes": 256000000,
+        #   "nvme_bytes": 0
+        # },
+        # "error": null,
+        # "status": 0
+        output = json.loads(self.result.stdout)
```

Comment:
Do we want to always return json format? Maybe we keep the if condition? I'd prefer the output from these test utilities to look the exact same as if running dmg directly on cmd line. I plan to use these modules to verify dmg json output and non json output are as expected (and the same). @mjmac  @phender

Reply:
I would really prefer to not perpetuate these dependencies on stable plaintext output in the python test harness. We already have unit/integration tests in the control plane for verifying that dmg's pretty printers work as expected for various outputs, and it's much easier to keep those tests in sync with any human-consumable output updates. In my opinion, it would be a better use of test engineering time to work on migrating the harness to consume JSON in order to reduce test fragility and maintenance overhead.